DuckDuckGo/GestureToolBarButton.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  BookmarksButton.swift
+//  DuckDuckGo
+//
+//  Copyright © 2019 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import UIKit
+
+protocol GestureToolBarButtonDelegate: NSObjectProtocol {
+    
+    func singleTapHandler()
+    func longPressHandler()
```

Comment:
Generally, for delegate design pattern we should add info which object is the sender of a message, example:
`func singleTapDetected(in: GestureToolBarButton)`

This way, in case we are using multiple GestureToolBarButtons, we can differentiate between them in delegate methods.